### ISSUE TITLE
Inequality functions dont return correct results with lang tags

### ISSUE THREAD

**Issue by pawanrawal** [dgraph-io/dgraph#960]
The following query on goldendata returns a name like `Prisoners` which shouldn't be in the result dataset.
```
{
  me(func: gt(name@en, "Steven Spielberg"), first: 15) {
    name@en
  }
}
```
 This is probably happening because while getting inequality tokens, we just go through all index keys irrespective of language and get uids. Now we might have compared against some other language (say `@ru`) but now we return value in `@en` which might not still be valid.

Another thing I noticed here is that even though we just want first 15, we go through the whole index. This might be unavoidable if there is an `orderasc` or `orderdesc` specified but if its not maybe we could stop the lookup as soon as we find 15 unique uids for the given language.

**Comment by tzdybal** [comment]
As we're going to filter language in a separate step, there is no way to avoid iteration over entire index.

**Comment by tzdybal** [comment]
Merged to release branch (03d8cc379f9472e2a966a4d8c1b6de75f96c4196).
